membership-common/src/main/scala/com/gu/memsub/subsv2/reads/SubJsonReads.scala

```diff
@@ -61,38 +60,21 @@ object SubJsonReads {
       (__ \ "specificBillingPeriod").readNullable[Int] and
       (__ \ "endDateCondition").read[EndDateCondition] and
       (__ \ "upToPeriods").readNullable[Int] and
-      (__ \ "upToPeriodsType").readNullable[UpToPeriodsType]
-  )(RatePlanCharge.apply _)
+      (__ \ "upToPeriodsType").readNullable[UpToPeriodsType] and
+      (__ \ "chargedThroughDate").readNullable[LocalDate] and
+      (__ \ "effectiveStartDate").read[LocalDate] and
+      (__ \ "effectiveEndDate").read[LocalDate]
```

Comment:
these fields we read from each charge as we go rather than separately grabbing them on the old line 71-82